crates/dbsp/src/circuit/runtime.rs

```diff
@@ -404,14 +412,14 @@ impl RuntimeInner {
 
     fn pin_cpu(&self) {
         if !self.pin_cpus.is_empty() {
-            let worker_index = Runtime::worker_index();
+            let local_worker_offset = Runtime::local_worker_offset();
```

Comment:
This may break the compiler: the constant operators generate their values only in worker 0.
They need to do the same in a distributed setting.
There is a need for a "global worker offset"

Reply:
Runtime::worker_index() is a global worker index. It will be 0 only on exactly one worker among all the hosts.

Runtime::local_worker_offset() is a local worker index. It will be 0 in one worker per host.

Do you mean that some compiler-generated code that currently uses the former should use the latter?

Reply:
Yes, it does